src/hera/workflows/_runner/script_annotations_util.py

```diff
@@ -63,6 +70,21 @@ def get_annotated_input_param(
     )
 
 
+def load_param_input(
+    param_value: str,
+    func_param_annotation: type,
+    loader: Optional[Callable[[str], Any]],
+) -> Any:
+    if loader is None:
+        return param_value
+
+    loaded_value = loader(param_value)
+    actual_type = unwrap_annotation(func_param_annotation)
+    if not isinstance(loaded_value, actual_type):
```

Comment:
What if `actual_type` doesn't have an `isinstance` check, e.g. if it's a `TypedDict`, or a `list[str]`? Do we need to do this validation for the user?

Reply:
Makes sense. We actually have pydantic doing the type validation for us. I've added a couple of extra tests for these scenarios.